Ff111 More OPFS related updaes

hamishwillee opened the pull request:
FF111 added support for OPFS in https://bugzilla.mozilla.org/show_bug.cgi?id=1811001. I added the main API in #18907 but it never occurred to me that the rest of the file system around this was not supported. So as per https://bugzilla.mozilla.org/show_bug.cgi?id=1811001#c11 this updates the interfaces in the File System Access API to indicate they are also supported.

- FileSystemWritableFileStream
- FileSystemDirectoryHandle
- FileSystemFileHandle
- FileSystemFileHandle
- FileSystemSyncAccessHandle


Note, WritableStream is already marked as  supported from FF100

@saschanaz Can you confirm this issue delivers full support for the API / that this is the right set of things to mark as supported in FF111?

saschanaz commented on api/FileSystemHandle.json:
```diff
@@ -235,7 +235,7 @@
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "111"
```
Not covered

saschanaz commented on api/DataTransferItem.json:
```diff
@@ -96,7 +96,7 @@
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "111"
```
Not covered

saschanaz commented on api/Window.json:
```diff
@@ -5713,7 +5713,7 @@
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "111"
```
Not covered

hamishwillee replied:
Reverted

saschanaz commented on api/Window.json:
```diff
@@ -5793,7 +5793,7 @@
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "111"
```
Not covered

saschanaz commented on api/Window.json:
```diff
@@ -5829,7 +5829,7 @@
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "111"
```
Not covered

saschanaz commented on api/FileSystemHandle.json:
```diff
@@ -162,7 +162,7 @@
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "111"
```
Not covered

saschanaz reviewed (changes requested):
When doing this kind of check, I highly recommend searching for the pref in searchfox inside dom/webidl: https://searchfox.org/mozilla-central/search?q=dom.fs.enabled&path=dom%2Fwebidl&case=true&regexp=false

Per the result DataTransferItem and Window members are not covered (as they are not part of https://fs.spec.whatwg.org/)

And it seems `spec_url`s are missing too, can you add them while you are at it?

saschanaz commented on api/FileSystemHandle.json:
```diff
@@ -199,7 +199,7 @@
             "edge": "mirror",
             "firefox": {
-              "version_added": false
+              "version_added": "111"
```
I'm not sure whether this even exists in any spec... Anyway, not covered

hamishwillee replied:
No, still a proposal https://github.com/whatwg/fs/pull/9

saschanaz reviewed (commented):
(Submitted a bit early.)

hamishwillee commented:
Hi @saschanaz 

Thanks so much - good tip on the searchfox. 
- I've dropped the commits for the DataTransferItem and Window changes.
- Reverted `FileSystemHandle.requestPermission` and queryPermission (along with status)
- `FileSystemHandle.remove()` reverted

All of these seem to have spec-urls unless they are non-standard - such as `FileSystemHandle.remove()`.
`FileSystemSyncAccessHandle` does omit spec urls for the sync-versions feature. Not sure where would be the right links for those.

I think we're good.

saschanaz commented on api/FileSystemHandle.json:
```diff
@@ -11,7 +11,7 @@
           "edge": "mirror",
           "firefox": {
-            "version_added": false
+            "version_added": "111"
```
LGTM, but it seems Gecko supports one more function on FileSystemHandle: `move()`. Should it added here? https://searchfox.org/mozilla-central/rev/3002762e41363de8ee9ca80196d55e79651bcb6b/dom/webidl/FileSystemHandle.webidl#16-22

hamishwillee replied:
Yes, done in https://github.com/mdn/browser-compat-data/pull/19056/files#r1131816623

saschanaz reviewed (commented):
Thanks! One comment:

hamishwillee commented on api/FileSystemHandle.json:
```diff
@@ -113,6 +113,39 @@
           }
         }
       },
+      "move": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/FileSystemHandle/move",
+          "support": {
+            "chrome": {
+              "version_added": false
+            },
+            "chrome_android": "mirror",
+            "edge": "mirror",
+            "firefox": {
+              "version_added": "111"
+            },
+            "firefox_android": "mirror",
+            "ie": {
+              "version_added": false
+            },
+            "oculus": "mirror",
+            "opera": "mirror",
+            "opera_android": "mirror",
+            "safari": {
+              "version_added": false
+            },
+            "safari_ios": "mirror",
+            "samsunginternet_android": "mirror",
+            "webview_android": "mirror"
+          },
+          "status": {
+            "experimental": true,
+            "standard_track": false,
+            "deprecated": false
+          }
+        }
+      },
```
@queengooborg FYI support for this was added in FF111 in https://bugzilla.mozilla.org/show_bug.cgi?id=1811001. However appears to be explicitly non-standard at this point as it is in a "PR" state - see the big banner when you access the URL https://whatpr.org/fs/10.html#api-filesystemhandle

Upshot, I've added it because that makes the compatibility story easier to track later, but there is no spec URL and it is marked as experimental and non-standard.

@saschanaz Added with above caveats ^^^. I don't plan to add documentation for this right now; any idea on when this PR is likely to go in?

saschanaz replied:
I have no idea, maybe @janvarga knows?

hamishwillee replied:
@saschanaz - cool - not urgent - just useful so we get a ping when it becomes standard.

BTW, thanks for all your help and advice on this. I will do better next time.

saschanaz replied:
No problem, you're welcome 👍

hamishwillee commented:
@queengooborg I think this can merge now.
